include/z64collision_check.h

```diff
@@ -9,245 +9,391 @@
-    /* 0x0C */ ColliderTrisItemInit* list;
-} ColliderTrisInit_Set3; // size = 0x10
+    /* 0x0C */ ColliderTrisElementInit* elements;
+} ColliderTrisInitType1; // size = 0x10
 
-typedef struct {
+typedef struct ColliderQuadDim {
     /* 0x00 */ Vec3f quad[4];
     /* 0x30 */ Vec3s dcMid; // midpoint of vectors d, c
     /* 0x36 */ Vec3s baMid; // midpoint of vectors b, a
-    /* 0x3C */ f32 unk_3C;
+    /* 0x3C */ f32 acDist; // distance to nearest AC collision this frame.
 } ColliderQuadDim; // size = 0x40
 
-typedef struct {
+typedef struct ColliderQuadDimInit {
     /* 0x00 */ Vec3f quad[4];
 } ColliderQuadDimInit; // size = 0x30
 
-typedef struct {
+typedef struct ColliderQuadElement {
+    /* 0x00 */ ColliderInfo info;
+    /* 0x24 */ ColliderQuadDim dim;
+} ColliderQuadElement; // size = 0x68
+
+typedef struct ColliderQuad {
     /* 0x00 */ Collider base;
-    /* 0x18 */ ColliderBody body;
+    /* 0x18 */ ColliderInfo info;
     /* 0x40 */ ColliderQuadDim dim;
 } ColliderQuad; // size = 0x80
 
-typedef struct {
+typedef struct ColliderQuadInit {
     /* 0x00 */ ColliderInit base;
-    /* 0x08 */ ColliderBodyInit body;
+    /* 0x08 */ ColliderInfoInit info;
     /* 0x20 */ ColliderQuadDimInit dim;
 } ColliderQuadInit; // size = 0x50
 
-typedef struct {
-    /* 0x00 */ ColliderInit_Set3 base;
-    /* 0x08 */ ColliderBodyInit body;
+typedef struct ColliderQuadInitType1 {
+    /* 0x00 */ ColliderInitType1 base;
+    /* 0x08 */ ColliderInfoInit info;
     /* 0x20 */ ColliderQuadDimInit dim;
-} ColliderQuadInit_Set3; // size = 0x50
+} ColliderQuadInitType1; // size = 0x50
 
-typedef struct {
+typedef struct OcLine {
     /* 0x00 */ Linef line;
-    /* 0x18 */ u16 unk_18;
+    /* 0x18 */ u16 ocFlags;
 } OcLine; // size = 0x1C
 
+typedef enum ColliderType {
+    COLTYPE_HIT0, // Blue blood, white hitmark
+    COLTYPE_HIT1, // No blood, dust hitmark
+    COLTYPE_HIT2, // Green blood, dust hitmark
+    COLTYPE_HIT3, // No blood, white hitmark
+    COLTYPE_HIT4, // Water burst, no hitmark
+    COLTYPE_HIT5, // No blood, red hitmark
+    COLTYPE_HIT6, // Green blood, white hitmark
+    COLTYPE_HIT7, // Red blood, white hitmark
+    COLTYPE_HIT8, // Blue blood, red hitmark
+    COLTYPE_METAL,
+    COLTYPE_NONE,
+    COLTYPE_WOOD,
+    COLTYPE_HARD,
+    COLTYPE_TREE
+} ColliderType;
+
+typedef enum ColliderShape {
+    COLSHAPE_JNTSPH,
+    COLSHAPE_CYLINDER,
+    COLSHAPE_TRIS,
+    COLSHAPE_QUAD,
+    COLSHAPE_INVALID
+} ColliderShape;
+
+/* Affects the sound Link's sword makes when hitting it, hookability,
+ * and possibly other things. It's definitely not flags, as all checks
+ * are == or !=. Will probably need more actors decomped to truly 
+ * understand what this is.
+ */
+typedef enum ElementType {
+    ELEMTYPE_UNK0,
+    ELEMTYPE_UNK1,
+    ELEMTYPE_UNK2,
+    ELEMTYPE_UNK3,
+    ELEMTYPE_UNK4,
+    ELEMTYPE_UNK5,
+    ELEMTYPE_UNK6,
+    ELEMTYPE_UNK7
+} ElementType;
+
+
+#define AT_OFF 0 // Cannot have AT collisions when set as AT
```

Comment:
There are usually no defines for "no flags set", however this case is a bit particular so it's probably fine. But I think they should all be called `_NONE` or something, like you did for OCLINE, to really emphasize on the fact that they mean "nothing enabled" (and not just the opposite of the `_ON` flag for instance).

Reply:
AC_OFF exists as a separate flag because of Bongo and Fishing (thanks Morita), which have colliders that start off.

Reply:
im confused here actually, isnt it supposed to be the opposite of the `on` flag? calling it `off` seems fine to me

Reply:
oh no i understand. the whole field is 0 so its supposed to be none as in "no flags"